core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/add-answer-group-modal.component.html

```diff
@@ -9,16 +9,20 @@ <h3>Add Response</h3>
       <strong>If the learner's answer...</strong>
     </div>
 
-    <rule-editor modal-id="modalId" rule="tmpRule" is-editable="EditabilityService.isEditable()" is-editing-rule-inline="false">
-    </rule-editor>
+    <oppia-rule-editor class="rule-editor-add-answer"
+                       [modalId]="modalId"
+                       [rule]="tmpRule"
+                       [isEditable]="isEditable"
+                       [isEditingRuleInline]="false">
+    </oppia-rule-editor>
 
     <br>
 
-    <div ng-if="!feedbackEditorIsOpen && !isLinearWithNoFeedback(tmpOutcome)"
+    <div *ngIf="!feedbackEditorIsOpen && true"
```

Comment:
Why replace `isLinearWithNoFeedback` with true?

Reply:
isLinearWithNoFeedback(tmpOutcome) function is not defined in `add-answer-group-modal.controller.ts` file. `&& true` removed from line no.21

Reply:
Oh, that's not great.. do we know when this was introduced?

Also, I think, if it is not used, then `!feedbackEditorIsOpen` should be sufficent.

Reply:
https://user-images.githubusercontent.com/77006897/177304033-49c237b1-707d-448c-a62b-5879f71fbb15.mp4



`All the interaction have feedback and for the interaction that don't have feedback like endExploration and continue the modal does not open for them. + we don't have defining of these function earlier` . please press `.`  and check in github web editor I have removed the  `&& true` from line no. 21.

Reply:
I think my first question still holds, do we know when isLinearWithNoFeedback was introduced in the HTML (e.g. which PR)?

Reply:
This was introduced 5 years ago in YogeshSharma [PR](https://github.com/oppia/oppia/pull/3942) in [add_answer_group_modal_directive.html file](https://github.com/1995YogeshSharma/oppia/blob/ba23f9cfcb40431f1234a596f1da32d14eb28feb/core/templates/dev/head/pages/exploration_editor/editor_tab/add_answer_group_modal_directive.html).

Reply:
Thanks @heyimShivam. It looks like the PR moves the addAnswerGroup modal in [state_editor_responses.html](https://github.com/oppia/oppia/pull/3942/files#diff-cf0f3af36cc723d6dec373f57556d1b5d8fab862076605c05c7dae6288fa8114) into a separate template [add_answer_group_modal_directive.html](https://github.com/oppia/oppia/pull/3942/files#diff-393950aa2a3882849cb787be0e0c56f43079b6c2c76ac55769b6850409833313) but it did not account for the availability of function `isLinearWithNoFeedback` in the new template. Based on this and the discussion above, it makes sense to just remove the function in the template. Thanks again for providing sufficient proof to validate your findings and investigating it!